std/machines/arith.asm

```diff
@@ -127,6 +128,53 @@ machine Arith with
         0
     };
 
+    let quotient_hint = query || if is_ec_operation() == 0 {
+        let y2 = y2_int();
+        let y3 = y3_int();
+        let x1 = x1_int();
+        let dividend = y2 << 256 + y3;
+        let quotient = divident / x1;
+        quotient
+    } else {
+        0
+    };
+
+    col witness quotient_0 query Query::Hint(fe(select_limb(quotient_hint(), 0)));
+    col witness quotient_1 query Query::Hint(fe(select_limb(quotient_hint(), 1)));
+    col witness quotient_2 query Query::Hint(fe(select_limb(quotient_hint(), 2)));
+    col witness quotient_3 query Query::Hint(fe(select_limb(quotient_hint(), 3)));
+    col witness quotient_4 query Query::Hint(fe(select_limb(quotient_hint(), 4)));
+    col witness quotient_5 query Query::Hint(fe(select_limb(quotient_hint(), 5)));
+    col witness quotient_6 query Query::Hint(fe(select_limb(quotient_hint(), 6)));
+    col witness quotient_7 query Query::Hint(fe(select_limb(quotient_hint(), 7)));
+    col witness quotient_8 query Query::Hint(fe(select_limb(quotient_hint(), 8)));
+    col witness quotient_9 query Query::Hint(fe(select_limb(quotient_hint(), 9)));
+    col witness quotient_10 query Query::Hint(fe(select_limb(quotient_hint(), 10)));
+    col witness quotient_11 query Query::Hint(fe(select_limb(quotient_hint(), 11)));
+    col witness quotient_12 query Query::Hint(fe(select_limb(quotient_hint(), 12)));
+    col witness quotient_13 query Query::Hint(fe(select_limb(quotient_hint(), 13)));
+    col witness quotient_14 query Query::Hint(fe(select_limb(quotient_hint(), 14)));
+    col witness quotient_15 query Query::Hint(fe(select_limb(quotient_hint(), 15)));
```

Comment:
The quotient is `y2`, right? I don't think you need new witness columns. I think you can do the same thing you've done for the quotient here but for `y2` and you're done :) The changes below should not be necessary, if I'm not mistaken.

The only thing is that the way it's written now, it will have a hint of `0` if `is_ec_operation() != 0`. I think instead it should be no hint, so something like:
```
col witness quotient_0 query match is_ec_operation() {
    0 => Query::Hint(fe(select_limb(quotient_hint(), 0))),
    _ => Query::None
};
```
(probably makes sense to pull that into a function, but you get the idea)

Reply:
> The only thing is that the way it's written now, it will have a hint of `0` if `is_ec_operation() != 0`

I know it's like that for the other hints (for `s`, for example), but the difference is that `s` is only used for EC operations and when it is, it always has a hint. So setting it to `0` doesn't hurt. In contrast, `y2` is also used by EC operations, but it should most likely not be 0!

Reply:
> The quotient is `y2`, right? I don't think you need new witness columns. I think you can do the same thing you've done for the quotient here but for `y2` and you're done :) The changes below should not be necessary, if I'm not mistaken.
> 
> The only thing is that the way it's written now, it will have a hint of `0` if `is_ec_operation() != 0`. I think instead it should be no hint, so something like:
> 
> ```
> col witness y2_0 query match is_ec_operation() {
>     0 => Query::Hint(fe(select_limb(quotient_hint(), 0))),
>     _ => Query::None
> };
> ...
> ```
> 
> (probably makes sense to pull that into a function, but you get the idea)

I assume you meant that the quotient is `y1`, because `y2` is the upper half of the 512 bit result? 

Replacing existing witness column with hint value makes sense to me. I'm just concerned by the scenario where the user commits `y1`. Will that be replaced by the hints?

Reply:
> > The only thing is that the way it's written now, it will have a hint of `0` if `is_ec_operation() != 0`
> 
> I know it's like that for the other hints (for `s`, for example), but the difference is that `s` is only used for EC operations and when it is, it always has a hint. So setting it to `0` doesn't hurt. In contrast, `y2` is also used by EC operations, but it should most likely not be 0!

Makes sense on this and I assume `Query::None` basically won't affect any existing commit values.

Reply:
> I assume you meant that the quotient is `y1`, because `y2` is the upper half of the 512 bit result?

Yes, sorry!

> Makes sense on this and I assume `Query::None` basically won't affect any existing commit values.

Yes, that answers your other question, right? `Query::None` means there is just no hint, so for the EC operations, everything is as before.

Also, if there were any constraints that contradict the hint, witgen would fail! It still checks that all constraints are satisfied.